File: 0509-fibonacci-number/509.py
class Solution:
    def fib(self, n: int) -> int:
        if (n <= 1):
            return n

        prev, prev2prev = 1, 0

        for _ in range(2, n + 1):
            curr = prev + prev2prev
            prev2prev = prev
            prev = curr

        return prev
        

File: 0509-fibonacci-number/509.c


int fib(int n) {
  if (n <= 1)
    return n;
  int prev      = 1;
  int prev2prev = 0;

  for (int i = 2; i <= n; i++) {
    int curr  = prev2prev + prev;
    prev2prev = prev;
    prev      = curr;
  }
  return prev;
}
